### PR TITLE
Added a nil check SecCertificateCopySubjectSummary

### LINKED ISSUE #249: Crash when trying to release NULL
We've got another crash coming from TrustKit

It looks like it's related to these lines.
https://github.com/datatheorem/TrustKit/blob/master/TrustKit/Pinning/ssl_pin_verifier.m#L66-L68

`SecCertificateCopySubjectSummary` can return NULL but nothing is checking for it before calling release.

```
Exception Type: EXC_BREAKPOINT (SIGTRAP)
Crashed Thread: 5

Application Specific Information:
*** CFRelease() called with NULL *** > boolValue > containsObject: > hashSubjectPublicKeyInfoFromCertificate: > objectForKeyedSubscript: > timeIntervalSinceReferenceDate

Thread 5 Crashed:
0   CoreFoundation                  0x315fb022c         CFRelease.cold.1
1   CoreFoundation                  0x315e2c534         CFRelease
2   TrustKit                        0x1045c79e0         verifyPublicKeyPin (ssl_pin_verifier.m:68)
3   TrustKit                        0x1045ced10         -[TSKPinningValidator evaluateTrust:forHostname:] (TSKPinningValidator.m:126)
4   TrustKit                        0x1045cf0e0         -[TSKPinningValidator handleChallenge:completionHandler:] (TSKPinningValidator.m:202)
```

### PR DESCRIPTION
**Pull request by CraigSiemens**
Fixes #249

### CODE REVIEW

File: TrustKit/Pinning/ssl_pin_verifier.m [hunk 1]
```diff
@@ -63,9 +63,12 @@ TSKTrustEvaluationResult verifyPublicKeyPin(SecTrustRef serverTrust, NSString *s
...
         // Extract the certificate
         SecCertificateRef certificate = SecTrustGetCertificateAtIndex(serverTrust, i);
+        
         CFStringRef certificateSubject = SecCertificateCopySubjectSummary(certificate);
-        TSKLog(@"Checking certificate with CN: %@", certificateSubject);
-        CFRelease(certificateSubject);
+        if (certificateSubject != nil) {
+            TSKLog(@"Checking certificate with CN: %@", certificateSubject);
+            CFRelease(certificateSubject);
+        }
```
Comment (nabla-c0d3) [inline TrustKit/Pinning/ssl_pin_verifier.m:71]: Lets also log something like `TSKLog(@"Could not parse certificate subject");`

**Review by nabla-c0d3 — CHANGES_REQUESTED** [review]
Thanks for the fix! Just left a minor request.

**Comment by CraigSiemens** [comment]
@nabla-c0d3 Could you take a look at this again when you have the chance?

**Comment by nabla-c0d3** [comment]
Thank you!

**Pull request merged**